pepper-apis/src/main/java/org/broadinstitute/ddp/service/KitCheckService.java

```diff
@@ -63,123 +62,162 @@ public KitCheckService(int batchSize) {
      * <p>Beyond these minimum requirements, the kit configuration also specifies additional criteria that are evaluated
      * before a participant qualifies for a kit request.
      *
-     * @param apisHandle the apis database handle
      * @return a mapping between each study and the number of participants who had a kit queued in this iteration. The
      *         mapping is <b>not</b> the total number of participants with kits. It's just the number of newly queued
      *         participants.
      */
-    public KitCheckResult checkForInitialKits(Handle apisHandle) {
+    public KitCheckResult checkForInitialKits() {
         KitCheckResult kitCheckResult = new KitCheckResult();
 
-        KitScheduleDao kitScheduleDao = apisHandle.attach(KitScheduleDao.class);
-        DsmKitRequestDao kitRequestDao = apisHandle.attach(DsmKitRequestDao.class);
-        KitConfigurationDao kitConfigurationDao = apisHandle.attach(KitConfigurationDao.class);
+        Queue<KitConfiguration> kitConfigs = withAPIsTxn(handle -> new ArrayDeque<>(
```

Comment:
Using a `Queue` so we remove things as we go instead of storing the whole list in memory the whole time. `ArrayDeque` seems to be a fast implementation of a queue, and it's more memory-efficient since it doesn't need to do any copies during a `remove()`.

Reply:
Maybe this approach of using limit and offset could be packaged up into a utility method that you give it the query and it gives you a query?

Reply:
Perhaps. But at first glance the tricky thing seems to be how we do order things. Some queries might order by `user_id`, while others should be ordered by something totally different. Not sure if that can be made generic. Also, having a utility for limit/offset seems a bit overkill, maybe it's just me.